configs: Allow GPU-FS to use HBMCtrl memory controller

TomXia opened the pull request:
This commit adds a command line argument --hbm-ctrl to enable HBMCtrl in GPU-FS. HBM_2000_4H_1x64 is tested.

mattsinc commented on configs/example/gpufs/Disjoint_VIPER.py:
```diff
@@ -107,8 +107,17 @@ def create(self, options, system, piobus, dma_devices):
         for mem_ctrl in gpu_mem_ctrls:
-            gpu_abstract_mems.append(mem_ctrl.dram)
+            # memctrl
+            if hasattr(mem_ctrl, "dram"):
```
instead of adding a second memory controller for the new HBM, shouldn't we add 2 dram's if the new model exists?  I guess this code seems a little complex to me, but maybe there is a good reason.  But it seems like this:

```
if (hbm_ctrl)
  gpu_abstract_mems.append(mem_ctrl.dram)
  gpu_abstract_mems.append(mem_ctrl.dram_2)
else if (!hbm_ctrl but using a different hbm)
  gpu_abstract_mems.append(mem_ctrl.dram)
else
  gpu_abstract_mems.append(mem_ctrl)
```
  
is more straightforward, no?

TomXia replied:
I was thinking about making it config-agnostic. Effectively it detects which controller is being used and adds the memory to system, without having to specify in args or change config (tested with MemCtrl, HBMCtrl and SimpleMemory). But, of course, I am happy to change it to a more direct way.

mattsinc replied:
@abmerop thoughts?

abmerop replied:
That is fine with me, although I believe we want to try to move to stdlib for most things (except GPU / X86 right now due to lack of >3GB memory)

mattsinc replied:
ok, we can leave it be then

mattsinc commented on configs/ruby/GPU_VIPER.py:
```diff
@@ -669,23 +669,51 @@ def construct_gpudirs(options, system, ruby_system, network):
-            xor_low_bit,
-        )
+
+        if options.hbm_ctrl:
```
please add a comment about why we are adding 2 controllers here -- it's obvious to you, but less obvious to others trying to use this

mattsinc reviewed (changes requested):
A couple minor things, but otherwise looks reasonable -- thanks!

abmerop reviewed (dismissed):
Is there a related change needed in stdlib files or does that already support this?

mattsinc commented:
@TomXia please address my other question and @abmerop stdlib question

TomXia committed:
configs: Allow GPU-FS to use HBMCtrl memory controller

This commit adds a command line argument --hbm-ctrl to enable HBMCtrl in GPU-FS.
HBM_2000_4H_1x64 is tested.

TomXia commented:
@abmerop :The stdlib already supports this. All we need is to update GPU configs.
@mattsinc :Added comments explaining the HBM config.

Thanks for reviewing!

mattsinc reviewed (approved):
Will wait for @abmerop to ok before merging
